hugr-py/src/hugr/__init__.py

```diff
@@ -9,4 +9,4 @@
 
 def get_serialisation_version() -> str:
     """Return the current version of the serialization schema."""
-    return "v2"
+    return "v3"
```

Comment:
I don't see any version checks on the `hugr-py` deserialisation. Should we throw an "unsupported version" error there?

It also seems like `Hugr.to_serial` is using a hard-coded "v1" tag instead of calling this function.
https://github.com/CQCL/hugr/blob/a2f34b6ea668dfb233066456e1ea85ff8c1f977c/hugr-py/src/hugr/hugr.py#L562-L568

Reply:
> It also seems like Hugr.to_serial is using a hard-coded "v1" tag instead of calling this function.

removed